hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ByteArrayObjectDataOutput.java

```diff
@@ -249,15 +246,19 @@ public void writeShort(int position, int v, ByteOrder byteOrder) throws IOExcept
 
     @Override
     public void writeUTF(final String str) throws IOException {
-        if (utfBuffer == null) {
-            utfBuffer = new byte[UTF_BUFFER_SIZE];
+        int len = (str != null) ? str.length() : NULL_ARRAY;
+        writeInt(len);
+        if (len > 0) {
+            ensureAvailable(len * 3);
```

Comment:
Is utf-8 maximum 4 bytes or 3 bytes? My understanding from https://en.wikipedia.org/wiki/UTF-8, it may be 4 bytes, can we double check on this?